modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesContextInformationManager.java

```diff
@@ -191,20 +207,51 @@ public void removeBinding(AndesBinding andesBinding) throws AndesException {
         notifyBindingListeners(andesBinding, BindingListener.BindingEvent.DELETED);
     }
 
-    private void notifyExchangeListeners(AndesExchange exchange, ExchangeListener.ExchangeChange change) throws AndesException {
-        for(ExchangeListener listener : exchangeListeners) {
+    /**
+     * Notifying the exchange listeners stating that a change has occurred for the exchanges in the
+     * local node. This will then get notified throughout the cluster if clustered deployment is
+     * available.
+     *
+     * @param exchange The andes exchange in which the change occurred.
+     * @param change   The change that is being occurred
+     * @throws AndesException
+     */
+    private void notifyExchangeListeners(AndesExchange exchange,
+                                         ExchangeListener.ExchangeChange change)
+            throws AndesException {
+        for (ExchangeListener listener : exchangeListeners) {
             listener.handleLocalExchangesChanged(exchange, change);
         }
     }
 
-    private void notifyQueueListeners(AndesQueue queue, QueueListener.QueueEvent change) throws AndesException {
-        for(QueueListener listener : queueListeners) {
+    /**
+     * Notifying the queue listeners stating that a change has occurred for the queues in the
+     * local node. This will then get notified throughout the cluster if clustered deployment is
+     * available.
+     *
+     * @param queue  Andes queue in which the change occurred.
+     * @param change The change that was occurred.
+     * @throws AndesException
+     */
+    private void notifyQueueListeners(AndesQueue queue, QueueListener.QueueEvent change)
+            throws AndesException {
+        for (QueueListener listener : queueListeners) {
             listener.handleLocalQueuesChanged(queue, change);
         }
     }
 
-    private void notifyBindingListeners(AndesBinding binding, BindingListener.BindingEvent change) throws AndesException {
-        for(BindingListener listener : bindingListeners) {
+    /**
+     * Notifying the bindings listeners stating that a change has occurred for the bindings in the
+     * local node. This will then get notified throughout the cluster if clustered deployment is
+     * available.
+     *
+     * @param binding The binding in which the change occurred.
+     * @param change  The change that occurred.
```

Comment:
The again would be more descriptive to define the possible types of bindings here